node/network/protocol/src/lib.rs

```diff
@@ -336,6 +337,40 @@ pub mod v1 {
 		Statement(Hash, SignedFullStatement)
 	}
 
+	#[derive(Debug, Clone, Copy, PartialEq, thiserror::Error)]
+	#[allow(missing_docs)]
+	pub enum CompressedPoVError {
+		#[error("Failed to compress a PoV")]
+		Compress,
+		#[error("Failed to uncompress a PoV")]
+		Uncompress,
+		#[error("Failed to decode the uncompressed PoV")]
+		Decode,
+	}
+
+	/// SCALE and GZip encoded [`PoV`].
+	#[derive(Debug, Clone, Encode, Decode, PartialEq)]
+	pub struct CompressedPoV(Vec<u8>);
+
+	impl CompressedPoV {
+		/// Create from the given [`PoV`].
+		pub fn from_pov(pov: &PoV) -> Result<Self, CompressedPoVError> {
+			let mut encoder = GzEncoder::new(Vec::with_capacity(1024), Compression::default());
+			pov.encode_to(&mut encoder);
+			encoder.finish().map(Self).map_err(|_| CompressedPoVError::Compress)
+		}
+
+		/// Uncompress, decode and return the [`PoV`].
+		pub fn as_pov(&self) -> Result<PoV, CompressedPoVError> {
```

Comment:
`as_` usually implies cheap conversion, whereas this should most likely be `to_`